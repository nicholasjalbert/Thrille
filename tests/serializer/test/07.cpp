//test cond_wait

#include <pthread.h>
#include <iostream>
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>

using namespace std;

int x;
bool y;
pthread_t child, child2;
pthread_mutex_t mutex, mutex2;
pthread_cond_t cond, cond2;

void setup() {
    pthread_mutex_init(&mutex, NULL);
    pthread_mutex_init(&mutex2, NULL);
    pthread_cond_init(&cond, NULL);
    pthread_cond_init(&cond2, NULL);
}

void teardown() {
    pthread_mutex_destroy(&mutex);
    pthread_mutex_destroy(&mutex2);
    pthread_cond_destroy(&cond);
    pthread_cond_destroy(&cond2);
}

void * routine1(void * arg) {
    y = true;
    pthread_cond_signal(&cond);
    x = 1;
}

int main(int argc, char *argv[]) {
    setup();
    y = false;
    x = 0;
    pthread_create(&child, NULL, routine1, NULL);
    pthread_mutex_lock(&mutex);
    while (!y) {
        pthread_cond_wait(&cond, &mutex);
    }
    pthread_mutex_unlock(&mutex);
    printf("x is %d\n", x);
    pthread_join(child, NULL);
    teardown();
}

