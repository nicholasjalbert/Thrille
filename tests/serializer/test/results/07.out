x is 0
Number of Context Switches: 4
Number of Preemptions: 1
